backend/prisma/schema.prisma

```diff
@@ -32,8 +54,111 @@ model Session {
 }
 
 model Profile {
-  id     String @id @default(uuid())
-  user   User @relation(fields: [userId], references: [id])
-  userId String @unique @default(uuid())
-  photoUrl String? 
-}
\ No newline at end of file
+  id        String   @id @default(uuid())
+  user      User     @relation(fields: [userID], references: [id])
+  userID    String   @unique
+  photoUrl  String?
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+}
+
+model Plan {
+  id               String       @id @default(uuid())
+  description      String
+  startDate        DateTime
+  endDate          DateTime
+  createdDate      DateTime     @default(now())
+  specialEventType EventType
+  contact          Contact?     @relation(fields: [contactID], references: [id])
+  contactID        String?      @unique
+  User             Pledge[]
+  createdAt        DateTime     @default(now())
+  updatedAt        DateTime     @updatedAt
+  GiftChoice       GiftChoice[]
+}
+
+model Pledge {
+  id           String        @id @default(uuid())
+  plan         Plan          @relation(fields: [planID], references: [id])
+  planID       String
+  user         User          @relation(fields: [userID], references: [id])
+  userID       String
+  inviteStatus InviteStatus?
+  role         Role
+  amount       Int?
+  pledgeStatus PledgeStatus?
+  createdAt    DateTime      @default(now())
+  updatedAt    DateTime      @updatedAt
+}
+
+enum PledgeStatus {
+  SENT_MONEY
+  HAVENT_SENT_MONEY
+}
+
+enum Role {
+  PLAN_LEADER
+  FRIEND
+}
+
+enum InviteStatus {
+  INVITED
+  RESENT_INVITE
+  ACCEPTED
+  DENY
+  NOT_APPLICABLE
+}
+
+model Contact {
+  id                 String                    @id @default(uuid())
+  email              String?
+  firstName          String
```

Comment:
Do we not need last name?

Reply:
Justin told me to leave this table for Fahad to figure it out 🙂

Reply:
fosho